Dockerfile

```diff
@@ -1,27 +1,52 @@
-FROM golang:1.10.3-alpine3.8
+FROM golang:1.15-alpine
+
+RUN apk add --update build-base git openssh
 
 ## github credentials
-RUN apk add --update bash curl git openssh py-pip
 ARG GIT_SSH_KEY
 RUN git config --global url.git@github.com:.insteadOf https://github.com/
 RUN mkdir ~/.ssh; ssh-keyscan -t rsa github.com > ~/.ssh/known_hosts
 RUN chmod -R 700 ~/.ssh; echo "${GIT_SSH_KEY}" > ~/.ssh/id_rsa; chmod 600 ~/.ssh/id_rsa
-RUN eval "$(ssh-agent -s)" && ssh-add ~/.ssh/id_rsa
+RUN eval "$(ssh-agent -s)" && \
+    ssh-add ~/.ssh/id_rsa
+
+ENV GOPATH=/go
+ENV BUILD_PATH=$GOPATH/src/github.com/Workiva/go-rest
+ENV V2=$BUILD_PATH/v2
+
+WORKDIR $BUILD_PATH
 
-#install glide
-RUN go get -u github.com/Masterminds/glide
+# v1
+COPY go.mod $BUILD_PATH/
+COPY go.sum $BUILD_PATH/
+COPY rest $BUILD_PATH/rest
 
-WORKDIR /go/src/github.com/Workiva/go-rest
-COPY . /go/src/github.com/Workiva/go-rest
+# v2
+COPY v2/go.mod $V2/
+COPY v2/go.sum $V2/
+COPY v2/rest $V2/rest
 
-# install dependencies
-RUN glide install
+# v1 - check formatting, build, test
+RUN go mod download
+WORKDIR $BUILD_PATH/rest
+RUN test -z $(gofmt -l .)
+RUN test -z $(go mod tidy -v)
```

Comment:
this works from /rest even though go.mod/go.sum is in the directory above? I think it makes more sense to run all the go commands from the directory with the module definition. `go fmt -l .` will need to be changed to `go fmt -l ./...` I think, but the rest should work I think